lib/adagrams.rb

```diff
@@ -0,0 +1,97 @@
+require "csv"
+require "set"
+
+def draw_letters
+  letters_hash = {
+    "A" => 9, "B" => 2, "C" => 2, "D" => 4, "E" => 12, "F" => 2,
+    "G" => 3, "H" => 2, "I" => 9, "J" => 1, "K" => 1, "L" => 4,
+    "M" => 2, "N" => 6, "O" => 8, "P" => 2, "Q" => 1, "R" => 6,
+    "S" => 4, "T" => 6, "U" => 4, "V" => 2, "W" => 2, "X" => 1,
+    "Y" => 2, "Z" => 1,
+  }
+
+  letters = []
+  letters_hash.each { |letter, frequency|
+    frequency.times do
+      letters << letter
+    end
+  }
+
+  return letters.sample(10)
+end
+
+def uses_available_letters?(input, letters_in_hand)
+  letters_in_hand = letters_in_hand.dup
+  input.each_char { |c|
+    index = letters_in_hand.index(c.upcase)
+    if index.nil?
+      return false
+    end
+
+    letters_in_hand.delete_at(index)
+  }
+  return true
+end
+
+def score_word(word)
+  word = word.upcase
+  score = (word.length >= 7) ? 8 : 0
+
+  score_chart = [
+    {:letters => ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"], :score => 1},
+    {:letters => ["D", "G"], :score => 2},
+    {:letters => ["B", "C", "M", "P"], :score => 3},
+    {:letters => ["F", "H", "V", "W", "Y"], :score => 4},
+    {:letters => ["K"], :score => 5},
+    {:letters => ["J", "X"], :score => 8},
+    {:letters => ["Q", "Z"], :score => 10},
+  ]
```

Comment:
Very interesting and cool data structure! It helps give a relationship between the letters and the score, while keeping it flexible in case letters need to move around. It feels a little more difficult to read compared to a just a hash (rather than your array of hashes), but this is still a great solution

Reply:
Hi Dee! Thank you for the feedback. I agree with you it is better to save the letter/score pairs in a hash. I wanted to save some keystrokes. Would something like this work?

hash = {}
%w("a", "b", "c"). each { |letter| hash[letter] = 1 }
%w("q", "z"). each { |letter| hash[letter] = 10 }

Or is there a better way?

Reply:
I think that's a great, more readable solution! I like it :)